### ISSUE TITLE
About data synchronization delay

### ISSUE THREAD

**Issue by lgy1027** [yandex-cloud/k8s-csi-s3#102]
Hello, I would like to ask you how to set the synchronization time when I mount a bucket to a pod and add or delete files on the page or directly in the pod. Because I tested it, the delay is still quite high. Does anyone know? What is the optimization direction?

**Comment by vitalif** [comment]
Hi.
What time do you mean?
Do you mean the time which passes before the pod sees new files uploaded directly to S3 in other applications?

**Comment by lgy1027** [comment]
> Hi. What time do you mean? Do you mean the time which passes before the pod sees new files uploaded directly to S3 in other applications?

Yes, for example, if I create a file in a pod or upload a bucket file directly to S3 on a page, is there any file synchronization time?

**Comment by vitalif** [comment]
There is no fixed delay, i.e. files are uploaded as quickly as possible, but writing doesn't wait for upload. You should call fsync if you want synchronous upload
https://github.com/yandex-cloud/geesefs/?tab=readme-ov-file#fsync
